crypto/box/src/hkdf_box.rs

```diff
@@ -0,0 +1,167 @@
+use crate::traits::{CryptoBox, Error};
+
+use aead::{
+    generic_array::{
+        sequence::Concat,
+        typenum::{IsLessOrEqual, Sum, Unsigned, B1},
+        ArrayLength, GenericArray,
+    },
+    Aead, Error as AeadError, NewAead,
+};
+use core::{convert::TryFrom, marker::PhantomData, ops::Add};
+use digest::{BlockInput, Digest, FixedOutput, Input, Reset};
+use hkdf::Hkdf;
+use keys::{GetBytes, Kex, KexSecret, PublicKey};
+use rand_core::{CryptoRng, RngCore};
+
+/// Represents an implementation of Ristretto-Box using Hkdf<Blake2b> and Aes128Gcm
+///
+/// This structure contains the actual cryptographic primitive details, and
+/// specifies part of the wire format of the "footer" where the ephemeral
+/// public key comes first, and the mac comes second.
+///
+/// Note: If when instantiating this, the IsLessOrEqual bounds fail,
+/// it may indicate that your Kex algorithm does not supply enough entropy to properly
+/// drive the Aead algorithm that you have chosen, and so the combination would be insecure.
+pub struct HkdfBox<KexAlgo, DigestAlgo, AeadAlgo>
```

Comment:
Rewriting the CryptoBox implementation to be generic in this way (over Kex and Aead) has been a longstanding goal, that was blocked in part by lack of `Ristretto: Kex`.